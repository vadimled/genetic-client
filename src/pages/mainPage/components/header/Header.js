import React, { Component } from "react";
import style from "./Header.module.scss";
import Notes from "Pages/mainPage/components/notes";
// import PropTypes from "prop-types";

class Header extends Component {
  render() {
    return (
      <div className={style["header-wrapper"]}>
        <div className="flex justify-start flex-row">
          <div className="left-wrapper">
            {/* TODO: Logo place*/}
            LOGO
          </div>
          <div className="left-wrapper">
            {/* TODO: Test ID place*/}
            <Notes valueNotes={"Test ID place..."}/>
          </div>
        </div>
        <div className="flex justify-start flex-row">
          <div className="right-wrapper">
            {/* TODO: Notification place*/}
            NT
          </div>
          <div className="right-wrapper">
            {/* TODO: User profile place*/}
            USR
          </div>
        </div>
      </div>
    );
  }
}

// Header.propTypes = {};

export default Header;
