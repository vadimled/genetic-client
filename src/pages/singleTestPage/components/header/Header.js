import React, { Component, Fragment } from "react";
import style from "./Header.module.scss";
import HeaderIcon from "GenericComponents/headerIcon";
import { ReactComponent as NotificationIcon } from "Assets/notifications.svg";
import { ReactComponent as InfoIcon } from "Assets/info.svg";
import User from "Pages/singleTestPage/components/header/components/user";
import {
  getTestId,
  getTumorInfoMode,
  getVariantPageTestId
} from "Store/selectors";
import { setTumorInfoMode } from "Actions/testActions";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import { ROUTES } from "Utils/constants";

class Header extends Component {
  constructor(props) {
    super(props);
    this.isMVP = process.env.REACT_APP_MVP_CONFIG === "true";
  }

  isVariantPage = () => {
    const regex = RegExp(ROUTES.isVariantPageRegex);
    return regex.test(this.props.location.pathname);
  };

  isSingleTestPage = () => {
    const regex = RegExp(ROUTES.isSingleTestRegex);
    return regex.test(this.props.location.pathname);
  };

  handelNotification = e => {
    console.log(e.target);
  };

  handelInfo = () => {
    const { showTumorInfo, setTumorInfoMode } = this.props;
    setTumorInfoMode(!showTumorInfo);
  };

  render() {
    console.log(this.props.location.pathname);
    return (
      <div className={style["header-wrapper"]}>
        <div className="flex justify-start flex-row">
          <div className="left-wrapper">
            {/* TODO: Logo place*/}
            LOGO
          </div>
          <div className="left-wrapper">
            {this.isVariantPage() && (
              <Link
                to={`/tests/${this.props.testId}`}
                data-testid={`go-back-button`}
                id={`go-back-button`}
                className={"go-back-button"}
              >
                <div className="go-back-button-text">{`< Go back`}</div>
              </Link>
            )}
          </div>
          <div className="left-wrapper">{this.props.testId}</div>
        </div>
        <div className="flex justify-start flex-row">
          <div className="right-side-item">
            <HeaderIcon
              isActive={this.props.showTumorInfo}
              customClassName={"info"}
              icon={<InfoIcon />}
              handelOnClick={this.handelInfo}
            />
          </div>
          {!this.isMVP && (
            <Fragment>
              <div className="right-side-item">
                <HeaderIcon
                  // isActive
                  customClassName={"notification"}
                  icon={<NotificationIcon />}
                  handelOnClick={this.handelNotification}
                />
              </div>

              <div className="right-side-item">
                <User
                  userName={"vadim malckin"}
                  avatarUrl={`https://randomuser.me/api/portraits/men/34.jpg`}
                />
              </div>
            </Fragment>
          )}
        </div>
      </div>
    );
  }
}

// Header.propTypes = {};

const mapStateToProps = state => {
  return {
    showTumorInfo: getTumorInfoMode(state),
    testId: getVariantPageTestId(state) || getTestId(state)
  };
};

function mapDispatchToProps(dispatch) {
  return {
    setTumorInfoMode: data => dispatch(setTumorInfoMode(data))
  };
}

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Header)
);
