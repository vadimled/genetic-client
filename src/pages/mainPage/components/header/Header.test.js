import React from "react";
import { renderWithRedux } from "Utils/test_helpers";
import "jest-dom/extend-expect";
import Header from "./Header";

it("renders", () => {
  const { asFragment } = renderWithRedux(<Header />);
  expect(asFragment()).toMatchSnapshot();
});
