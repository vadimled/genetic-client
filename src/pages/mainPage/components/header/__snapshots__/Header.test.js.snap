// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot 1`] = `
<DocumentFragment>
  <div
    class="header-wrapper"
  >
    <div
      class="flex justify-start flex-row"
    >
      <div
        class="left-wrapper"
      >
        LOGO
      </div>
      <div
        class="left-wrapper"
      />
    </div>
    <div
      class="flex justify-start flex-row"
    >
      <div
        class="right-side-item"
      >
        <div
          class="header-icon-wrapper"
        >
          <div
            class="icon info"
            data-testid="header-icon-info"
          >
            <svg>
              info.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="right-side-item"
      >
        <div
          class="header-icon-wrapper"
        >
          <div
            class="icon notification"
            data-testid="header-icon-notification"
          >
            <svg>
              notifications.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="right-side-item"
      >
        <div
          class="user-wrapper"
        >
          <picture
            class="avatar"
          >
            <img
              alt="User's avatar"
              src="https://randomuser.me/api/portraits/men/34.jpg"
            />
          </picture>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
